v2/snmp/manager.go

```diff
@@ -2,19 +2,25 @@ package snmp
 
 import (
 	"context"
+	"encoding/asn1"
+	"fmt"
 	"net"
+	"strconv"
+	"strings"
+
+	"github.com/geoffgarside/ber"
 )
 
 // Manager provides an interface for SNMP device management.
 type Manager interface {
 	// Issues an SNMP GET request for the specified oids.
-	Get(ctx context.Context, oids []string) (*Response, error)
+	Get(ctx context.Context, oids []string) (*PDU, error)
 
 	// Issues an SNMP GET NEXT request for the specified oids.
-	GetNext(ctx context.Context, oids []string) (*Response, error)
+	GetNext(ctx context.Context, oids []string) (*PDU, error)
 
 	// Issues an SNMP GET BULK request for the specified oids.
-	GetBulk(ctx context.Context, oids []string, nonRepeaters int, maxRepetitions int) (*Response, error)
+	GetBulk(ctx context.Context, oids []string, nonRepeaters int, maxRepetitions int) (*PDU, error)
```

Comment:
worth adding a comment to explain what nonRepeaters does?  rest are obvious.

Reply:
Yes, but I'll probably just add a reference to the rfc, where the semantics of nonRepeaters and maxRepetitions are described in some detail.

Reply:
Added appropriate links to RFC1905.